### PR TITLE
Better support for Armenian language

### PR DESCRIPTION
**Pull request by amcoder36**
Added better support for Armenian language

- Corrected punctuation
- Added spell check support

The word list was generated from https://hy.wikisource.org/, the content is licensed under a [Creative Commons Attribution-ShareAlike 3.0 Unported license](https://creativecommons.org/licenses/by-sa/3.0/).

### CODE REVIEW

File: app/src/main/java/org/dslul/openboard/inputmethod/latin/spellcheck/AndroidWordLevelSpellCheckerSession.java [hunk 1]
```diff
@@ -62,6 +64,15 @@ public abstract class AndroidWordLevelSpellCheckerSession extends Session {
     private static final String quotesRegexp =
             "(\\u0022|\\u0027|\\u0060|\\u00B4|\\u2018|\\u2018|\\u201C|\\u201D)";
 
+    private static final Map<Integer, String> scriptToPunctuationRegexMap = new TreeMap<>();
+
+    static {
+        scriptToPunctuationRegexMap.put(
```
Comment (amcoder36) [inline app/src/main/java/org/dslul/openboard/inputmethod/latin/spellcheck/AndroidWordLevelSpellCheckerSession.java:70]: There needs to be a way to ignore language specific punctuation in addition to the standard English ones. See the usage below.

File: tools/make-keyboard-text/src/main/resources/values-hy/donottranslate-more-keys.xml [hunk 2]
```diff
@@ -23,30 +23,35 @@
          U+0532: "Բ" ARMENIAN CAPITAL LETTER BEN
          U+0533: "Գ" ARMENIAN CAPITAL LETTER GIM -->
     <string name="keylabel_to_alpha">&#x0531;&#x0532;&#x0533;</string>
-    <!-- U+055E: "՞" ARMENIAN QUESTION MARK
```
Comment (amcoder36) [inline tools/make-keyboard-text/src/main/resources/values-hy/donottranslate-more-keys.xml:26]: Corrected the punctuation for Armenian, left comments in Armenian and English where it's translatable to English.

**Comment by MajeurAndroid** [comment]
Thanks for your contribution! 

- Can you add english descriptions in [tools/make-keyboard-text/src/main/resources/values-hy/donottranslate-more-keys.xml](tools/make-keyboard-text/src/main/resources/values-hy/donottranslate-more-keys.xml), every unicode character has one. For example U+055B do have an english description, which is "Armenian Emphasis Mark", you can find it [here](https://www.compart.com/fr/unicode/U+055B).
- And also can you clarify a bit why you changed morekeys_punctuation characters ? As well as inverted exclamation and question marks removal ? 
- Can you add a TODO comment that mentions to genrify language specific punctuation later in [app/src/main/java/org/dslul/openboard/inputmethod/latin/spellcheck/AndroidWordLevelSpellCheckerSession.java](app/src/main/java/org/dslul/openboard/inputmethod/latin/spellcheck/AndroidWordLevelSpellCheckerSession.java#L283).

**Comment by amcoder36** [comment]
Thanks for the review @MajeurAndroid 

1) and 3) I addressed and pushed. To answer 2)

- Also can you clarify a bit why you changed morekeys_punctuation characters?
  - The current versions is a mixture of Armenian and English punctuations when Armenian is selected. One good example is the questions mark where the Armenian version is `՞` but it will display `?` and `՞` in the same list, where first is never used in Armenian language and adds confusion.
  - Wanted to get it to a point that only punctuates that are actually used in Armenian language (and the full set) is displayed in the more list.
  - I tried to order it in a way that more frequently used ones are closer to the more button/the finger for more convenient one hand typing.
As well as inverted exclamation and question marks removal ?
- There are no inverted exclamation and question marks in Armenian. 

Hope those answer your question. Let me know if more context is needed.

**Comment by amcoder36** [comment]
@MajeurAndroid please let me know if more needs to be addressed, or it's good to be merged.

**Comment by amcoder36** [comment]
> @amcoder36 sorry for the late answer ! Thanks for the modifications. Regarding inverted exclamation and question marks, even if they are not used in one language they are often offered as an extra option for bilingual users. Removing them can bother some people that always exepcted them. Can you put them back ? After that we'll be ready to merge :) Thanks !

Thank you for the input @MajeurAndroid. I added those back.

**Pull request merged**